### PR TITLE
feat: support right-to-left languages

### PR DESCRIPTION
**Pull request by iainmcgin**
This makes the Ray component styles sensitive to direction, such that components will render in the
expected orientation for right-to-left languages. Ideally, this would have been done using CSS' new
Logical Properties, but support for these is not pervasive yet, so style overrides are applied using
parent selectors wherever a left-right asymmetrical style is applied. The examples have been
extended to show every component with Hebrew right-to-left rendering.

BREAKING CHANGE: Component modifiers that ended in "left" or "right" have been changed to "start"
and "end" respectively, to communicate the alignment relative to the expected flow of text in the
host language. All examples have been updated, but any clients using Ray will likely need to be
updated. Alternatively, we could temporarily alias make "left" == "start" and "right" == "end"
semantically, until we can fully deprecate those styles.

### CODE REVIEW

File: packages/core/src/components/text-field/_text-field.scss [hunk 1]
```diff
@@ -123,12 +123,12 @@
 
       .#{$ray-class-prefix}text-field__label,
       .#{$ray-class-prefix}text-area__label {
-        padding-top: $ray-field-v-spacing-compact;
+        padding-block-start: $ray-field-v-spacing-compact;
```
Comment (cmugla) [inline packages/core/src/components/text-field/_text-field.scss]: this isn't supported in IE or Edge:

Screen Shot 2019-05-22 at 7 01 12 PM

Comment (maniator) [reply packages/core/src/components/text-field/_text-field.scss]: It is however supported in the edge canary ;-)

**Comment by adamraider** [comment]
Looks like CI did not run, could you try pushing an empty commit?

It also might be because you're trying to merge from a forked repo. Not sure

**Comment by adamraider** [comment]
> Alternatively, we could temporarily alias make "left" == "start" and "right" == "end"
semantically, until we can fully deprecate those styles.

I would like to do this :)

File: packages/core/src/components/text-field/_text-field.scss (hunk 1 above)
Comment (i8ramin) [reply packages/core/src/components/text-field/_text-field.scss]: Does `padding-top` have an impact on right-to-left languages?

**Comment by i8ramin** [comment]
This project is pretty old, but found in some initial research on the topic a while back -- https://github.com/twitter/css-flip. Not sure if helpful in anyway, but thought I'd share

File: packages/core/src/components/breadcrumb/_breadcrumb.scss [hunk 2]
```diff
@@ -20,13 +20,24 @@
...
 
+      [dir='rtl'] & {
+        margin-right: 0;
+        margin-left: 0.5rem;
+      }
+
       &::after {
         margin-left: 0.5rem;
-        content: '➝';
+        content: '→';
```
[the remark this answers is not in the document]
Comment (iainmcgin) [reply packages/core/src/components/breadcrumb/_breadcrumb.scss]: Unfortunately there is no flipped variant of this arrow in the Dingbats unicode range: The character prior to this change was U+279D "Triangle-Headed Rightwards Arrow". The new character is U+2192 "Rightwards Arrow", which comes from the "Arrows" Unicode block (https://www.compart.com/en/unicode/block/U+2190) that has the equivalent U+2190 "Leftwards Arrow".

Are you concerned about a specific stylistic difference in the arrows? And if so, are you OK with the left arrow for RTL looking different from the right arrow for LTR?

File: packages/core/src/components/image/_image.scss [hunk 3]
```diff
@@ -53,6 +53,10 @@ $ray-image-ratio-map: (
   -webkit-line-clamp: 2;
   -webkit-box-orient: vertical;
   margin-top: $ray-spacing-xs;
+
+  [dir='rtl'] & {
+    text-align: left;
```
[the remark this answers is not in the document]
Comment (iainmcgin) [reply packages/core/src/components/image/_image.scss]: I was just mirroring the choice to right-align text in captions in LTR, meaning we would left-align text in captions in RTL. Quite a mental gymnastic workout we're getting here, at least we don't _also_ have to deal with top-to-bottom, right-to-left traditional Japanese script 😆

File: packages/core/src/components/text-field/_text-field.scss (hunk 1 above)
Comment (iainmcgin) [reply packages/core/src/components/text-field/_text-field.scss]: Ah my bad, this is a remnant from my first attempt to do what was needed using logical properties. The future "correct" / pedantic way to do everything in future once logical properties are standardized (they're still draft right now) will be to drop all ideas of top / bottom / left / right in favor of block-start / block-end / inline-start / inline-end.

Anyway, it doesn't impact just supporting RTL. I'll revert this.

**Comment by iainmcgin** [comment]
> > Alternatively, we could temporarily alias make "left" == "start" and "right" == "end"
> > semantically, until we can fully deprecate those styles.
> 
> I would like to do this :)

I've reintroduced the left and right labeled rules as aliases of start and end, PTAL.

File: packages/core/src/components/breadcrumb/_breadcrumb.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmugla) [reply packages/core/src/components/breadcrumb/_breadcrumb.scss]: Yes, stylistic difference. You have to flip it using css transform

File: packages/core/src/components/image/_image.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (cmugla) [reply packages/core/src/components/image/_image.scss]: ha, yet! yeah makes sense, good 👁

**Comment by adamraider** [comment]
@iainmcgin Can you try pushing your branch directly to the repo? For some reason CI isn't running.

File: packages/core/src/components/text-field/_text-field.scss (hunk 1 above)
Comment (cmugla) [reply packages/core/src/components/text-field/_text-field.scss]: TIL!

File: packages/core/src/components/button/_button.scss [hunk 4]
```diff
@@ -29,14 +29,37 @@
...
+      [dir='rtl'] & {
+        margin-right: 0;
+        margin-left: 1rem;
+      }
     }
 
-    &--pull-left {
+    // --pull-left is DEPRECATED in favor of --pull-start, and will be
+    // removed in a future release of Ray. Use of pull-left may result
+    // in confusing behavior for RTL language sensitive layouts.
```
Comment (cmugla) [inline packages/core/src/components/button/_button.scss]: 👏

Comment (cmugla) [reply packages/core/src/components/button/_button.scss]: love these comments, thank you

Comment (adamraider) [reply packages/core/src/components/button/_button.scss]: This class is undocumented and I plan on removing it all together anyway lol

**Comment by iainmcgin** [comment]
Moved to #110, closing this.

File: packages/core/src/components/breadcrumb/_breadcrumb.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (iainmcgin) [reply packages/core/src/components/breadcrumb/_breadcrumb.scss]: Sneaky 😆 applying this in #110

**Pull request closed without merging**